service/history/api/create_workflow_util.go

```diff
@@ -86,6 +91,29 @@ func NewWorkflowWithSignal(
 		return nil, err
 	}
 
+	// This workflow is created for the hsm attached to it and will not generate actual workflow tasks.
+	// This is a bit of a hacky way to distinguish between a "real" workflow and a top level state machine.
+	// This code will change as the scheduler HSM project progresses.
+	hsmOnlyWorkflow := startRequest.StartRequest.WorkflowType.Name == scheduler.WorkflowType && shard.GetConfig().UseExperimentalHsmScheduler(startRequest.NamespaceId)
```

Comment:
Old and new scheduler workflows should be able to coexist.. they should use different types and the only thing that should check the feature flag is the starter (frontend)

Reply:
I know it's experimental but using a different type is even easier than what you have here, isn't it? Why not just do it now and not worry about changing it later?

My understanding is that there's still a lot of value in having schedules run as an hsm inside a workflow, we don't need to get to top-level hsms for this to be useful.